### ISSUE TITLE
support struct sort

### ISSUE THREAD

**Issue by hawkingrei** [zhangyunhao116/pdqsort#2]
Hello, TiDB has a plan to use pdqsort to improve sort performance. but we find that Golang will support pdqsort in the Go1.19. So we decide to use this library to achieve our goals. however, we find this library only sorts this [base type](https://github.com/zhangyunhao116/pdqsort/blob/master/constraint.go) and we need to sort the slice of custom type. Have you a plan to improve it?

**Comment by zhangyunhao116** [comment]
Happy to see TiDB plan to use the new implementation!
The generic version of pdqsort is located in https://github.com/golang/exp/tree/master/slices for now, and it will be moved to the `std` in the future. The problem is that the APIs in the `slices` may be changed at any time. A possible solution is to copy the related code into the TiDB project, once the `slices` is merged into `std`, we can easily change the code.
For this repo, I'm still not sure whether it needs to be maintained, since it has already been merged into the Go runtime.

**Comment by hawkingrei** [comment]
Thank you for your suggestion. I have started to use ```https://github.com/golang/exp/tree/master/slices ``` to implement [it](https://github.com/pingcap/tidb/pull/34404).
